test/networkd-test.py

```diff
@@ -461,16 +452,14 @@ def test_search_domains_too_long(self):
 
         subprocess.check_call(['systemctl', 'start', 'systemd-networkd'])
 
-        if os.path.islink('/etc/resolv.conf'):
-            for timeout in range(50):
-                with open('/etc/resolv.conf') as f:
-                    contents = f.read()
-                if 'search one\n' in contents:
-                    break
-                time.sleep(0.1)
-            self.assertIn('search %(p)s0 %(p)s1 %(p)s2 %(p)s3\n'
-                          '# Total length of all search domains is too long, remaining ones ignored.' % {'p': name_prefix},
-                          contents)
+        for timeout in range(50):
+            with open(RESOLV_CONF) as f:
+                contents = f.read()
+            if 'search one\n' in contents:
+                break
+            time.sleep(0.1)
+        self.assertRegex(contents, 'search .*%(p)s0 %(p)s1 %(p)s2 %(p)s3\n'
+                      '# Total length of all search domains is too long, remaining ones ignored.' % {'p': name_prefix})
```

Comment:
Ah, the "openstacklocal" makes the string long enough so that the a...a3 already falls off the edge.. Going to fix that.

This particular check was previously skipped, which is why I'd rather move to testing the resolved generated output.